core/Db/TransactionLevel.php

```diff
@@ -69,6 +83,7 @@ public function restorePreviousStatus()
         if ($this->statusBackup) {
             $value = strtoupper($this->statusBackup);
             $this->statusBackup = null;
+            $this->db->supportsUncommitted = null;
```

Comment:
I'm not understanding why we would reset this one? This would clear the cache every time meaning the cache would effectively never be used?